src/ng/rootScope.js

```diff
@@ -1270,8 +1272,8 @@ function $RootScopeProvider() {
           // yes, this code is a bit crazy, but it works and we have tests to prove it!
           // this piece should be kept in sync with the traversal in $digest
           // (though it differs due to having the extra check for $$listenerCount)
-          if (!(next = ((current.$$listenerCount[name] && current.$$childHead) ||
-              (current !== target && current.$$nextSibling)))) {
+          if (!(next = ((current.$$listenerCount[name] && !stopPropagation && current.$$childHead) ||
+              (current !== target && ((stopPropagation = false) || current.$$nextSibling))))) {
```

Comment:
Maybe it should split that complicated expression into more `if`s?
But then it will be completely different than traversal in `$digest`